src/python_framework_v02/troute/nhd_io.py

```diff
@@ -1034,6 +1034,80 @@ def get_channel_restart_from_wrf_hydro(
     return q_initial_states
 
 
+def read_lite_restart(
+    file
+):
+    '''
+    Open lite restart pickle files. Can open either waterbody_restart or channel_restart
+    
+    Arguments
+    -----------
+        file (string): File path to lite restart file
+        
+    Returns
+    ----------
+        df (DataFrame): restart states
+        t0 (datetime): restart datetime
+    '''
+    
+    # open pickle file to pandas DataFrame
+    df = pd.read_pickle(pathlib.Path(file))
+    
+    # extract restart time as datetime object
+    t0 = df['time'].iloc[0].to_pydatetime()
+    
+    return df.drop(columns = 'time') , t0
+    
+
+def write_lite_restart(
+    q0, 
+    waterbodies_df, 
+    t0, 
+    restart_parameters
+):
+    '''
+    Save initial conditions dataframes as pickle files
+    
+    Arguments
+    -----------
+        q0 (DataFrame):
+        waterbodies_df (DataFrame):
+        t0 (datetime.datetime):
+        restart_parameters (string):
+        
+    Returns
+    -----------
+        
+    '''
+    
+    output_directory = restart_parameters.get('lite_restart_output_directory', None)
+    if output_directory:
+        
+        # create pathlib object for output directory
+        output_path = pathlib.Path(output_directory)
+        
+        # create restart filenames
+        t0_str = t0.strftime("%Y%m%d%H%M")
+        channel_restart_filename = 'channel_restart_'+t0_str
+        waterbody_restart_filename = 'waterbody_restart_'+t0_str
+        
+        q0_out = q0.copy()
+        q0_out['time'] = t0
+        q0_out.to_pickle(pathlib.Path.joinpath(output_path, channel_restart_filename))
+        LOG.debug('Dropped lite channel restart file %s' % pathlib.Path.joinpath(output_path, channel_restart_filename))
+
+        if not waterbodies_df.empty:
+            wbody_initial_states = waterbodies_df.loc[:,['qd0','h0']]
+            wbody_initial_states['time'] = t0
+            wbody_initial_states.to_pickle(pathlib.Path.joinpath(output_path, waterbody_restart_filename))
+            LOG.debug('Dropped lite waterbody restart file %s' % pathlib.Path.joinpath(output_path, channel_restart_filename))
+        else:
+            LOG.debug('No lite waterbody restart file dropped becuase waterbodies are either turned off or do not exist in this domain.')
+        
+    else:
+        LOG.error("Not writing lite restart files. No lite_restart_output_directory variable was not specified in configuration file.")
+    
+
```

Comment:
New function that does the writing-out of lite restart files as pickle. Restart data are contained in DataFrames, which when saved as pickle, can be conveniently read back in with all of the indexing and formatting maintained. A new column is appended to DataFrames containing channel and waterbody initial states that contains the restart time.